pythonforandroid/recipes/python3/__init__.py

```diff
@@ -219,6 +229,13 @@ def get_recipe_env(self, arch=None, with_flags_in_cc=True):
             warning('lld not found, linking without it. '
                     'Consider installing lld if linker errors occur.')
 
+        if 'MAKEFLAGS' in env.keys() and version_starts_with("3.11"):
+            if "-j" in env['MAKEFLAGS']:
+                __ = env['MAKEFLAGS'].split(" ")
+                __.pop(__.index("-j") + 1)  # -j 10, here pop 10
+                __.pop(__.index("-j"))  # here pop '-j'
+                env["MAKEFLAGS"] = " ".join(__)
```

Comment:
By performing some tests, this does not seem needed at all.
Can you still reproduce the issue by performing a clean build?
Can you please add some logs?

Reply:
Ignore this one. Now I recall we had a previous discussion about it.

This seems hacky however, I'm looking into CPython codebase to see if a bug / major change has been introduced.

Reply:
Actually I am unable to reproduce this now. So should I remove it?

App built with no modifications to MAKEFLAGS:
BUILD LOGS: https://transfer.sh/BPUNYQpaLU/build-log.txt

Logcat:
```
11-18 19:38:57.052 26259 26259 V pythonutil: Loading library: python3.11
11-18 19:38:57.063 26259 26259 V pythonutil: Loading library: main
11-18 19:38:57.065 26259 26259 V pythonutil: Loaded everything!
11-18 19:38:57.193 26259 26315 I python  : Initializing Python for Android
11-18 19:38:57.193 26259 26315 I python  : Setting additional env vars from p4a_env_vars.txt
11-18 19:38:57.193 26259 26315 I python  : Changing directory to the one provided by ANDROID_ARGUMENT
11-18 19:38:57.193 26259 26315 I python  : /data/user/0/org.test.myapp/files/app
11-18 19:38:57.193 26259 26315 I python  : Preparing to initialize python
11-18 19:38:57.193 26259 26315 I python  : _python_bundle dir exists
11-18 19:38:57.193 26259 26315 I python  : calculated paths to be...
11-18 19:38:57.193 26259 26315 I python  : /data/user/0/org.test.myapp/files/app/_python_bundle/stdlib.zip:/data/user/0/org.test.myapp/files/app/_python_bundle/modules
11-18 19:38:57.194 26259 26315 I python  : set wchar paths...
11-18 19:38:57.271 26259 26315 I python  : Initialized python
11-18 19:38:57.271 26259 26315 I python  : AND: Init threads
11-18 19:38:57.272 26259 26315 I python  : testing python print redirection
11-18 19:38:57.274 26259 26315 I python  : Android path ['.', '/data/user/0/org.test.myapp/files/app/_python_bundle/stdlib.zip', '/data/user/0/org.test.myapp/files/app/_python_bundle/modules', '/data/user/0/org.test.myapp/files/app/_python_bundle/site-packages']
11-18 19:38:57.275 26259 26315 I python  : os.environ is environ({'PATH': '/sbin:/system/sbin:/system/bin:/system/xbin:/odm/bin:/vendor/bin:/vendor/xbin', 'DOWNLOAD_CACHE': '/data/cache', 'ANDROID_BOOTLOGO': '1', 'ANDROID_ROOT': '/system', 'ANDROID_ASSETS': '/system/app', 'ANDROID_DATA': '/data', 'ANDROID_STORAGE': '/storage', 'EXTERNAL_STORAGE': '/sdcard', 'ASEC_MOUNTPOINT': '/mnt/asec', 'BOOTCLASSPATH': '/system/framework/core-oj.jar:/system/framework/core-libart.jar:/system/framework/conscrypt.jar:/system/framework/okhttp.jar:/system/framework/bouncycastle.jar:/system/framework/apache-xml.jar:/system/framework/ext.jar:/system/framework/framework.jar:/system/framework/telephony-common.jar:/system/framework/voip-common.jar:/system/framework/ims-common.jar:/system/framework/sprengine.jar:/system/framework/android.hidl.base-V1.0-java.jar:/system/framework/android.hidl.manager-V1.0-java.jar:/system/framework/knoxsdk.jar:/system/framework/timakeystore.jar:/system/framework/fipstimakeystore.jar:/system/framework/sec_edm.jar:/system/framework/knoxanalyticssdk.jar:/system/framework/smartbondingservice.jar:/system/framework/ucmopensslenginehelper.jar:/system/framework/esecomm.jar:/system/framework/securetimersdk.jar:/system/framework/sec_sdp_sdk.jar:/system/framework/sec_sdp_hidden_sdk.jar:/system/framework/framework-oahl-backward-compatibility.jar:/system/framework/android.test.base.jar:/system/framework/knoxvpnuidtag.jar:/system/framework/SemAudioThumbnail.jar:/system/framework/knoxguard.jar:/system/framework/drutils.jar', 'SYSTEMSERVERCLASSPATH': '/system/framework/services.jar:/system/framework/ethernet-service.jar:/system/framework/wifi-service.jar:/system/framework/com.android.location.provider.jar:/system/framework/hqm.jar:/system/framework/hcm.jar', 'MC_AUTH_TOKEN_PATH': '/efs', 'KNOX_STORAGE': '/data/knox/ext_sdcard', 'ENC_EMULATED_STORAGE_TARGET': '/storage/enc_emulated', 'ANDROID_SOCKET_zygote': '11', 'ANDROID_ENTRYPOINT': 'main.pyc', 'ANDROID_ARGUMENT': '/data/user/0/org.test.myapp/files/app', 'ANDROID_APP_PATH': '/data/user/0/org.test.myapp/files/app', 'ANDROID_PRIVATE': '/data/user/0/org.test.myapp/files', 'ANDROID_UNPACK': '/data/user/0/org.test.myapp/files/app', 'PYTHONHOME': '/data/user/0/org.test.myapp/files/app', 'PYTHONPATH': '/data/user/0/org.test.myapp/files/app:/data/user/0/org.test.myapp/files/app/lib', 'PYTHONOPTIMIZE': '2', 'P4A_BOOTSTRAP': 'SDL2', 'PYTHON_NAME': 'python', 'P4A_IS_WINDOWED': 'True', 'KIVY_ORIENTATION': 'Portrait', 'P4A_NUMERIC_VERSION': 'None', 'P4A_MINSDK': '21', 'LC_CTYPE': 'C.UTF-8'})
11-18 19:38:57.275 26259 26315 I python  : Android kivy bootstrap done. __name__ is __main__
11-18 19:38:57.275 26259 26315 I python  : AND: Ran string
11-18 19:38:57.275 26259 26315 I python  : Run user program, change dir and execute entrypoint
11-18 19:38:57.658 26259 26315 I python  : [ERROR  ] Error when copying logo directory
11-18 19:38:57.658 26259 26315 I python  : Traceback (most recent call last):
11-18 19:38:57.658 26259 26315 I python  :   File "/content/KivyTest/.buildozer/android/platform/build-armeabi-v7a/build/python-installs/myapp/armeabi-v7a/kivy/__init__.py", line 411, in <module>
11-18 19:38:57.658 26259 26315 I python  :   File "/content/KivyTest/.buildozer/android/platform/build-armeabi-v7a/build/other_builds/python3/armeabi-v7a__ndk_target_21/python3/Lib/shutil.py", line 561, in copytree
11-18 19:38:57.658 26259 26315 I python  :   File "/content/KivyTest/.buildozer/android/platform/build-armeabi-v7a/build/other_builds/python3/armeabi-v7a__ndk_target_21/python3/Lib/shutil.py", line 515, in _copytree
11-18 19:38:57.658 26259 26315 I python  : shutil.Error: [('/data/user/0/org.test.myapp/files/app/_python_bundle/site-packages/kivy/data/logo/kivy-icon-512.png', '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-512.png', "[Errno 13] Permission denied: '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-512.png'"), ('/data/user/0/org.test.myapp/files/app/_python_bundle/site-packages/kivy/data/logo/kivy-icon-16.png', '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-16.png', "[Errno 13] Permission denied: '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-16.png'"), ('/data/user/0/org.test.myapp/files/app/_python_bundle/site-packages/kivy/data/logo/kivy-icon-64.png', '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-64.png', "[Errno 13] Permission denied: '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-64.png'"), ('/data/user/0/org.test.myapp/files/app/_python_bundle/site-packages/kivy/data/logo/kivy-icon-24.png', '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-24.png', "[Errno 13] Permission denied: '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-24.png'"), ('/data/user/0/org.test.myapp/files/app/_python_bundle/site-packages/kivy/data/logo/kivy-icon-64.ico', '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-64.ico', "[Errno 13] Permission denied: '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-64.ico'"), ('/data/user/0/org.test.myapp/files/app/_python_bundle/site-packages/kivy/data/logo/kivy-icon-48.png', '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-48.png', "[Errno 13] Permission denied: '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-48.png'"), ('/data/user/0/org.test.myapp/files/app/_python_bundle/site-packages/kivy/data/logo/kivy-icon-32.png', '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-32.png', "[Errno 13] Permission denied: '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-32.png'"), ('/data/user/0/org.test.myapp/files/app/_python_bundle/site-packages/kivy/data/logo/kivy-icon-128.png', '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-128.png', "[Errno 13] Permission denied: '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-128.png'"), ('/data/user/0/org.test.myapp/files/app/_python_bundle/site-packages/kivy/data/logo/kivy-icon-256.png', '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-256.png', "[Errno 13] Permission denied: '/data/user/0/org.test.myapp/files/app/.kivy/icon/kivy-icon-256.png'"), ('/data/user/0/org.test.myapp/files/app/_python_bundle/site-packages/kivy/data/logo', '/data/user/0/org.test.myapp/files/app/.kivy/icon', "[Errno 13] Permission denied: '/data/user/0/org.test.myapp/files/app/.kivy/icon'")]
11-18 19:38:57.678 26259 26315 I python  : [WARNING] [Config      ] Older configuration version detected (0 instead of 27)
11-18 19:38:57.678 26259 26315 I python  : [WARNING] [Config      ] Upgrading configuration in progress.
11-18 19:38:57.679 26259 26315 I python  : [DEBUG  ] [Config      ] Upgrading from 0 to 1
11-18 19:38:57.687 26259 26315 I python  : [INFO   ] [Logger      ] Record log in /data/user/0/org.test.myapp/files/app/.kivy/logs/kivy_23-11-18_0.txt
11-18 19:38:57.693 26259 26315 I python  : [INFO   ] [Kivy        ] v2.2.1
11-18 19:38:57.694 26259 26315 I python  : [INFO   ] [Kivy        ] Installed at "/data/user/0/org.test.myapp/files/app/_python_bundle/site-packages/kivy/__init__.pyc"
11-18 19:38:57.694 26259 26315 I python  : [INFO   ] [Python      ] v3.11.5 (main, Nov 18 2023, 13:56:27) [Clang 14.0.6 (https://android.googlesource.com/toolchain/llvm-project 4c603efb0
11-18 19:38:57.695 26259 26315 I python  : [INFO   ] [Python      ] Interpreter at ""
11-18 19:38:57.696 26259 26315 I python  : [INFO   ] [Logger      ] Purge log fired. Processing...
11-18 19:38:57.697 26259 26315 I python  : [INFO   ] [Logger      ] Purge finished!
11-18 19:38:58.016 26259 26315 I python  : [INFO   ] [Factory     ] 190 symbols loaded
11-18 19:38:58.731 26259 26315 I python  : [INFO   ] [Image       ] Providers: img_tex, img_dds, img_sdl2 (img_pil, img_ffpyplayer ignored)
11-18 19:38:58.797 26259 26315 I python  : [INFO   ] [Text        ] Provider: sdl2
11-18 19:38:58.859 26259 26315 I python  : [INFO   ] [Window      ] Provider: sdl2
11-18 19:38:58.901 26259 26315 I python  : [INFO   ] [GL          ] Using the "OpenGL ES 2" graphics system
11-18 19:38:58.904 26259 26315 I python  : [INFO   ] [GL          ] Backend used <sdl2>
11-18 19:38:58.904 26259 26315 I python  : [INFO   ] [GL          ] OpenGL version <b'OpenGL ES 3.2 v1.r28p0-01rel0.###other-sha0123456789ABCDEF0###'>
11-18 19:38:58.905 26259 26315 I python  : [INFO   ] [GL          ] OpenGL vendor <b'ARM'>
11-18 19:38:58.905 26259 26315 I python  : [INFO   ] [GL          ] OpenGL renderer <b'Mali-T830'>
11-18 19:38:58.906 26259 26315 I python  : [INFO   ] [GL          ] OpenGL parsed version: 3, 2
11-18 19:38:58.907 26259 26315 I python  : [INFO   ] [GL          ] Texture max size <8192>
11-18 19:38:58.907 26259 26315 I python  : [INFO   ] [GL          ] Texture max units <16>
11-18 19:38:58.959 26259 26315 I python  : [INFO   ] [Window      ] auto add sdl2 input provider
11-18 19:38:58.961 26259 26315 I python  : [INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
11-18 19:39:00.633 26259 26315 I python  : [WARNING] [Base        ] Unknown <android> provider
11-18 19:39:00.634 26259 26315 I python  : [INFO   ] [Base        ] Start application main loop
11-18 19:39:00.643 26259 26315 I python  : [INFO   ] [GL          ] NPOT texture support is available
```

Reply:
Let's clean it up! This fix does not seem to be needed.